sdk/core/core-http/src/policies/proxyPolicy.ts

```diff
@@ -30,43 +29,48 @@ function loadEnvironmentProxyValue(): string | undefined {
   return httpsProxy || allProxy || httpProxy;
 }
 
-// Check whether the given `uri` matches the noProxyList. If it matches, any request sent to that same `uri` won't set the proxy settings.
+// Check whether the host of a given `uri` is in the noProxyList.
+// If there's a match, any request sent to the same host won't have the proxy settings set.
 function isBypassed(uri: string): boolean | undefined {
-  if (byPassedList.has(uri)) {
-    return byPassedList.get(uri);
+  if (noProxyList.length === 0) {
+    return false;
   }
-  loadNoProxy();
-  let isBypassedFlag = false;
   const host = URLBuilder.parse(uri).getHost()!;
-  for (const proxyString of noProxyList) {
-    if (proxyString[0] === ".") {
-      if (uri.endsWith(proxyString)) {
+  if (byPassedList.has(host)) {
+    return byPassedList.get(host);
+  }
+  let isBypassedFlag = false;
+  for (const pattern of noProxyList) {
+    if (pattern[0] === ".") {
+      // This should match either domain it self or any subdomain or host
+      // .foo.com will match foo.com it self or *.foo.com
```

Comment:
Is this behavior for NO_PROXY documented anywhere? (or any of the proxy environment variables for that matter?)

Reply:
We have a table listing supported environment variables at https://github.com/Azure/azure-sdk/blob/33b5bef4fd10f66140db74779892a06753b6b4f7/_includes/tables/environment_variables.md but I don't think we have details about each variables' behaviors.  This implementation is a direct port from .NET.

Reply:
Ah cool. I wonder how we can surface this table so users of our SDK can know about it (I had no idea this table existed, and it's even in a separate repo!) But that's really a separate issue...

Reply:
We published the table as part of [general Azure Core guidelines](https://azure.github.io/azure-sdk/general_azurecore.html#environment-variables) but I agree with you that it is not really accessible to end customers.  It could be part of, or referenced from a potential Azure Core doc about ProxyPolicy.

Reply:
We only have a couple docs in core-http. I logged https://github.com/azure/azure-sdk-for-js/issues/13561 for more improvement.